scripts/09-history-migration-to-mongo.py

```diff
@@ -0,0 +1,455 @@
+        self.family = family
+        self.backend_id = backend_id
+        self.path = path
+
+    def __hash__(self):
+        return hash((self.hostname, self.port, self.family, self.backend_id, self.path))
+
+    def __eq__(self, other):
+        return (self.hostname == other.hostname and
+                self.port == other.port and
+                self.family == other.family and
+                self.backend_id == other.backend_id and
+                self.path == other.path)
+
+    def dump(self):
+        return {
+            'hostname': self.hostname,
+            'port': self.port,
+            'family': self.family,
+            'backend_id': self.backend_id,
+            'path': self.path,
+        }
+
+    def __repr__(self):
+        return (
+            '<GroupNodeBackendRecord: '
+            'hostname={hostname}, '
+            'port={port}, '
+            'family={family}, '
+            'backend_id={backend_id}, '
+            'path={path}>'.format(
+                hostname=self.hostname,
+                port=self.port,
+                family=self.family,
+                backend_id=self.backend_id,
+                path=self.path
+            )
+        )
+
+
+class GroupCoupleRecord(object):
+    # TODO: Move constats to history module
+    def __init__(self, couple, timestamp=None, type='automatic'):
+        self.couple = couple
+        self.timestamp = timestamp
+        self.type = type
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __eq__(self, other):
+        return set(self.couple) == set(other.couple)
+
+    def dump(self):
+        return {
+            'couple': self.couple,
+            'type': self.type,
+            'timestamp': self.timestamp,
+        }
+
+    def __repr__(self):
+        return '<GroupCoupleRecord: couple {}, timestamp {}, type {}>'.format(
+            self.couple,
+            self.timestamp,
+            self.type
+        )
+
+
+def migrate_history(s, coll):
+
+    ip_hostname_cache = {}
+
+    def ip_to_hostname(ip_address):
+        if ip_address in ip_hostname_cache:
+            return ip_hostname_cache[ip_address]
+
+        try:
+            res = socket.gethostbyaddr(ip_address)[0]
+        except Exception as e:
+            print 'Failed to resolve {}: {}'.format(ip_address, e)
+            return ip_address
+
+        ip_hostname_cache[ip_address] = res
+        return res
+
+    def _unserialize(data):
+            group_state = msgpack.unpackb(data)
+            group_state['nodes'] = list(group_state['nodes'])
+            if 'couples' not in group_state:
+                group_state['couples'] = []
+            group_state['couples'] = list(group_state['couples'])
+            return group_state
+
+    def _node_state_type(node_state):
+        return node_state.get(
+            'type',
+            (
+                node_state.get('manual', False) and 'manual' or
+                'automatic'
+            )
+        )
```

Comment:
I have rewritten it more explicitly.
https://github.com/yandex/mastermind/commit/66e99d0b3655e08f7a60b3261e090a2f1c305040